app/controllers/kudos_controller.rb

```diff
@@ -5,9 +5,20 @@ class KudosController < ApplicationController
   skip_before_filter :store_location
 
   def index
-    @work = Work.find(params[:work_id])
-    @kudos = @work.kudos.includes(:pseud => :user).with_pseud
-    @guest_kudos_count = @work.kudos.by_guest.count
+    if params[:work_id]
+      @work = Work.find(params[:work_id])
+      @kudos = @work.kudos.includes(pseud: :user).with_pseud
+      @guest_kudos_count = @work.kudos.by_guest.count
+    end
+    if params[:user_id]
+      @user = User.find_by_login(params[:user_id])
+      @check_ownership_of = @user
+      check_ownership
+      # collext a list of pseuds the user may have left kudos under
+      pseuds = Pseud.where(user_id: @user.id).value_of(:id)
+      kudos_list = Kudo.where(pseud_id: pseuds).value_of(:commentable_id)
+      @kudos = Work.where(id: kudos_list).page(params[:page])
```

Comment:
And as a suggestion for improvement, it would be great if you could also search your kudos history for work title just as a simple query here--obviously more search capability would be even better, but even that small bit of search functionality would make this more useful by allowing someone to narrow down their list if they've left loads of kudos.